CSharpSourceCode/AttributeDataSystem/PartyAttributeManager.cs

```diff
@@ -0,0 +1,437 @@
+                TOWCommon.Say("Already added"); 
+            }
+            
+        }
+
+        public void DeregisterParty(MobileParty party, PartyBase partyBase)
+        {
+            if(_partyAttributes.ContainsKey(party.Party.Id))
+            {
+                _partyAttributes.Remove(party.Party.ToString());
+                //TOWCommon.Say("Removed " + party.Party.Id); 
+            }
+            else
+            {
+               // TOWCommon.Say("Couldn't remove " + party.Party.Id); 
+            }
+        }
+        private void OnGameLoaded()
+        {
+            TOWCommon.Say("save game restored with "+ _partyAttributes.Count + "parties in the dictionary");
+            _isloaded = true;
+            
+            
+            //for later: Check if Attributes are valid, reinitalize for parties if not
+            //TOWCommon.Say(GetAttribute(Campaign.Current.MainParty.Party.Id).LeaderAttribute.race);
+        }
+        
+        private Action OnGameSaving()
+        {
+            
+            return new Action(OnGameSaveAction);
+        }
+        
+        private void OnGameSaveAction()
+        {
+            TOWCommon.Say("save game stored with "+ _partyAttributes.Count + "parties in the dictionary");
+        }
+        
+        private void OnNewGameCreatedPartialFollowUpEnd(CampaignGameStarter campaignGameStarter)
+        {
+            InitalizeAttributes();
+            _isloaded = true;
+        }
+
+        private void FillWindsOfMagic(float TickValue)
+        {
+            foreach (var entry in _partyAttributes)
+            {
+                if (entry.Value.MagicUserParty)
+                    entry.Value.WindsOfMagic += TickValue;
+            }
+        }
+
+        private Action OnDailyTick()
+        {
+            return new Action(DailyMessage);
+        }
+
+        private void DailyMessage()
+        {
+            string text = "";
+            foreach (var culture in _cultureCounts)
+            {
+                text+=(culture.Key + " " + _cultureCounts[culture.Key]+ ", ");
+            }
+
+            text +=" Main player has WOM: "+ GetAttribute(Campaign.Current.MainParty.Party.Id).WindsOfMagic;
+            TOWCommon.Say(text);
+        }
+        
+        private void InitalizeAttributes()
+        {
+            var parties = Campaign.Current.MobileParties;
+
+            foreach (MobileParty party in parties)
+            {
+                if (_partyAttributes.ContainsKey(party.Party.Id))
+                {
+                    TOWCommon.Say(party.Id.ToString()+  " was already there");
+                    continue;
+                }
+                EnterPartyIntoDictionary(party);
+            }
+            TOWCommon.Say(_partyAttributes.Count + "of "+  parties.Count+ " were initalized");
+        }
+
+        public override void SyncData(IDataStore dataStore)
+        {
+            dataStore.SyncData("_partyAttributes", ref _partyAttributes);
+        }
+        
```

Comment:
Delete these new lines? In general, just be consistent with the number of newlines between different structures. e.g. delete the new line on 37, 82, 83... add a new line after 92, etc